### PR TITLE
{bio}[foss/2023a] Single-cell-python-bundle v2024.02

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/214

Requires:
- [x] #20814 
- [x] #20636
- [x] #20387
- [x] #20616
- [x] #20551
- [x] #20707
- [x] #20477
- [x] #20289
- [x] #20115
- [x] #20109
- [x] #20080
- [x] #20002
- [x] #20025
- [x] #20111
- [x] #20067
- [x] #19994
- [x] #19981
- [x] #19963

TODO:
- [x] homepage
- [x] description
- [x] scvi-tools.eb (requires https://github.com/easybuilders/easybuild-easyconfigs/pull/20119)
- [x] scArches.eb - required scvi-tools
- [x] scib-metrics.eb - required scvi-tools

Exts to add:
- [x] Anndata2ri
- [x] bbknn
- [x] cassiopeia-lineage
- [x] Sc_toolbox
- [x] Altair
- [x] Pertpy
   - [x] required scvi-tools
- [x] tascCODA
- [x] Liana-py 
- [x] Harmonypy
- [x] Scirpy
- [x] CellPhoneDB
- [x] Decoupler
- [x] contrastiveVI
   - should be part of scvi-tools now
- [x] Schist

### CODE REVIEW

File: easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,215 @@
+easyblock = 'PythonBundle'
+
+name = 'Single-cell-python-bundle'
+version = '2024.02'
+
+homepage = 'xxx'
```
Comment (boegel) [inline easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb:6]: use `(none)` here

File: easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+description = """Bundle of softwares for single-cell research ???"""
```
Comment (boegel) [inline easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb:7]:
```suggestion
description = """Bundle of tools for single-cell sequence analysis"""

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+name = 'Single-cell-python-bundle'
+version = '2024.02'
+
+homepage = (None)
+description = """Bundle of tools for single-cell sequence analysis."""
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+# builddependencies = [('poetry', '1.5.1')]
```
Comment (boegel) [inline easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb:11]: @pavelToman `poetry` is required for installing `pypi-latest`, so you'll have to uncomment this?

See failed test report in https://gist.github.com/boegelbot/caeab59f44f6611f00bff1c7073031c9

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
Fix for the fsspec error: https://github.com/easybuilders/easybuild-easyconfigs/pull/20477

File: easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        'checksums': ['3a99faf412691159c068b3ff01c15307ce2fd2cf6b860199434874e002040a84'],
+    }),
+    ('toyplot', '1.0.3', {
+        'checksums': ['7b7b2bc5784fd75e5c695300bffc80d568c83bebef543bb54e6e6c2229912edd'],
+    }),
+    ('toytree', '2.0.5', {
+        'checksums': ['7be04ca310067e0e9737449700d6ab1b68b4379b64e2d22f2f7697a70030ceb0'],
+    }),
+    ('tascCODA', '0.1.3', {
+        'preinstallopts': "sed -i '28d' setup.py && ",
```
Comment (boegel) [inline easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb:124]: @pavelToman Can you explain why this is needed?

It seems like you're just removing the requirement for `tensorflow-probability` here?

Comment (boegel) [reply easybuild/easyconfigs/s/Single-cell-python-bundle/Single-cell-python-bundle-2024.02-foss-2023a.eb:124]: OK, please add a comment in the easyconfig to explain that then.

And ideally we use a `sed` command that shows what gets removed exactly (so not using a random line number)

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 4 out of 5 (5 easyconfigs in total) cns1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/377161a87e9bfdc11f25e09571b3431e for a full test report.

Seems GDAL/3.7.1-foss-2023a is missing on generoso?

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3158.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/4cba265b4d6d703bae34ed679496de75 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @pavelToman!

**Pull request merged**